strapi/strapi
# [Blocks editor] Fix enter key behavior in code blocks

### What does it do?

Instead of relying on the default Slate behavior when the user hits the "enter" key, add a new paragraph below the block every time... except when the block has a custom `handleEnterKey` in the store. This PR only adds handleEnterKey for code blocks, to make sure hitting enter creates a new line within the block instead of creating a new block.

Handling lists will come in another PR.

### Why is it needed?

The default Slate behavior is often not the wanted one.

### How to test it?

Create a code block, hit enter when your cursor is in it, you should have a new line still inside the block.

Hit enter on any other type of block, you should get a paragraph block below your current block.
